packages/ui/src/InfoModalIcon.tsx

```diff
@@ -0,0 +1,41 @@
+import React, {FC, useState} from "react";
+import {Pressable} from "react-native";
+
+import {InfoModalIconProps} from "./Common";
+import {Heading} from "./Heading";
+import {Modal} from "./Modal";
+
+export const InfoModalIcon: FC<InfoModalIconProps> = ({
+  infoModalChildren,
+  infoModalSubtitle,
+  infoModalText,
+  infoModalTitle,
+}) => {
+  const [infoModalVisibleState, setInfoModalVisibleState] = useState(false);
+  return (
+    <>
+      <Modal
+        primaryButtonOnClick={() => setInfoModalVisibleState(false)}
+        primaryButtonText="Close"
+        size="md"
+        subtitle={infoModalSubtitle}
+        text={infoModalText}
+        title={infoModalTitle}
+        visible={infoModalVisibleState}
+        onDismiss={() => setInfoModalVisibleState(false)}
+      >
+        {infoModalChildren}
+      </Modal>
+      <Pressable
+        aria-role="button"
+        hitSlop={{top: 10, bottom: 10, left: 10, right: 10}}
+        style={{marginLeft: 8}}
+        onPress={() => setInfoModalVisibleState(true)}
+      >
+        <Heading color="secondaryLight" size="sm">
```

Comment:
is it possible to to consider the indexing for the styling?
if odd secondary light
if even white
the bg color of the button is blending with the bg color of the row

Reply:
Do you mean the icon to open the modal for each row in the DataTable? This is just used in the header, so it doesn't look like the background should ever change.

Reply:
Went ahead and added a custom component that follows this, looks nicer! Thanks!